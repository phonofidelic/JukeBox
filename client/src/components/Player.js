import React, { Component } from 'react';
import {Howl, Howler} from 'howler';
import PlayerProgress from './PlayerProgress';
import PlayerControls from './PlayerControls';
import QueueList from './QueueList';
import Grid from '@material-ui/core/Grid';
import Collapse from '@material-ui/core/Collapse';
import { withTheme } from '@material-ui/core/styles';

export class Player extends Component {
	render() {
		const { 
			player, 
			time,
			userAgentIsMobile,
			handleStopTrack,
			handlePlayTrack,
			handlePauseTrack,
			handlePlayNext,
			handlePlayPrev,
			handleToggleQueue,
			handlePlayFromQueue,
			handleSeek,
			theme,
		} = this.props;

		const styles = {
			root: {
			  position: 'fixed',
			  bottom: userAgentIsMobile ? theme.dimensions.nav.navHeight : 0, // Makes room for bottom Nav componenet
			  width: '100%',
			  boxShadow: '0px -1px 20px 1px #ccc',
			  backgroundColor: theme.palette.secondary.light,
			  borderBottom: `solid 1px ${theme.palette.primary.main}`,
			  padding: '0',
			  zIndex: 2,
			}
		};
		
		// [CONTRACT]
		// If player.queue has tracks: 
		// * render controls and queue list.
		// * pass player action handlers to PLayerControls component.
		// * render QueueList with queue and currentTrack props inside MUI Collapse component.
		// Otherwise: Player component is an empty div.
		return (
			<div style={styles.root}>		
				<PlayerProgress 
					player={player}
					handleSeek={handleSeek}
				 />
				<Grid container>
					<Grid item xs={12}>
						<PlayerControls 
							player={player}
							handleStopTrack={handleStopTrack}
							handlePlayTrack={handlePlayTrack}
							handlePauseTrack={handlePauseTrack}
							handlePlayNext={handlePlayNext}
							handlePlayPrev={handlePlayPrev}
							handleToggleQueue={handleToggleQueue}
						/>
					</Grid>
				</Grid>
				<Collapse direction="up" in={player.showQueue} collapsedHeight="0px">
					<QueueList 
						queue={player.queue} 
						queueIndex={player.queueIndex}
						currentTrack={player.currentTrack} 
						handleStopTrack={handleStopTrack}
						handlePlayTrack={handlePlayTrack}
						handlePlayFromQueue={handlePlayFromQueue}
					/>
				</Collapse>
			</div>
		);
	}
}

export default withTheme()(Player);
