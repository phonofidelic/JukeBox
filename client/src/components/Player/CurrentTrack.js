import React, { useContext } from 'react';
import styled from 'styled-components';

import { ThemeContext, getPlayerHeight } from '../../contexts/theme.context';

import Typography from '@material-ui/core/Typography';

const CurrentTrackContainer = styled.div`
	display: flex;
	text-align: left;
	height: ${getPlayerHeight}
`
const CurrentTrackImage = styled.div`

`

const CurrentTrackInfo = styled.div`
	width: ${props => props.playerIsOpen? '100%' : '120px'};
	padding: 8px;
`

const CurrentTrack = props => {
	const {
		player,
		playerIsOpen,
	} = props;

	const theme = useContext(ThemeContext);

	return (
		<CurrentTrackContainer 
			theme={theme}
			style={{height: theme.dimensions.player.height}}
		>
			<CurrentTrackImage theme={theme}>
				<img 
					src={player.currentTrack.image.src} 
					alt="Album art" 
					width={theme.dimensions.player.height} 
					height={theme.dimensions.player.height} 
				/>
			</CurrentTrackImage>
			<CurrentTrackInfo 
				theme={theme}
				playerIsOpen={playerIsOpen}
			>
				<Typography noWrap>{player.currentTrack.title}</Typography>
				<Typography noWrap variant="caption">{player.currentTrack.artist.name}</Typography>
			</CurrentTrackInfo>
		</CurrentTrackContainer>
	)
}

export default CurrentTrack;