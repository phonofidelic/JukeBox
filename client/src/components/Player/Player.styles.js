const styles = theme => ({
	root: {
	  position: 'fixed',
	  bottom: 0,
	  width: '100%',
	  zIndex: 1,
	},
	containerMobile: {
		backgroundColor: theme.palette.secondary.light,
		width: '100%',
		// height: theme.dimensions.player.height + 10,
		margin: '0 auto',
		boxShadow: '0px -1px 20px 1px #ccc',
	  padding: '0',
	},
	containerDesktop: {
		backgroundColor: theme.palette.secondary.light,
		maxWidth: theme.dimensions.libraryDesktop.maxWidth,
		margin: '0 auto',
		boxShadow: '0px -1px 20px 1px #ccc',
	  borderBottom: `solid 1px ${theme.palette.primary.main}`,
	  padding: '0',
	  zIndex: 3,
	},
});

export default styles;

