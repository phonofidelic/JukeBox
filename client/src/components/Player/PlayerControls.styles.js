const styles = theme => ({
	root: {
		border: '1px solid red'
	},
	currentTrackContainer: {
		textAlign: 'left',
    // padding: '5px',
	},
	toggleButtonContainer: {
		textAlign: 'right',
	},
});

export default styles;

