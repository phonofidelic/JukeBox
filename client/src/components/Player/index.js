import React, { Component } from 'react';
import styles from './Player.styles';
import PropTypes from 'prop-types';
import {
	ThemeContext,
	getSecondaryBackgroundColor,
} from '../../contexts/theme.context';

import Backdrop from '../Backdrop'

import PlayerBar from './PlayerBar';
import PlayerProgress from './PlayerProgress';
import PlayerControls from './PlayerControls';
import QueueList from './QueueList';

import Draggable from 'react-draggable';

import withStyles from '@material-ui/core/styles/withStyles';

const WINDOW_TOP = window.innerHeight * -1;
const TRIGGER_DRAG_DISTANCE = WINDOW_TOP / 3;
const DRAG_TRANSITION = 'all .5s ease';

export class Player extends Component {
	static contextType = ThemeContext;

	constructor(props) {
		super(props)
		this.state = {
			isOpen: false,
			isDragging: false,
			showQueue: true,
			position: 0,
			windowHeight: window.innerHeight,
			dragEndTransition: 'none',
		}
	}

	componentDidMount() {
		window.addEventListener('resize', this.handleWindowResize);
	}

	componentWillUnmount() {
		window.removeEventListener('resize', this.handleWindowResize);
	}

	handleWindowResize = e => {
		this.setState({windowHeight: e.target.innerHeight});
	}
	
	handleDragStart = (e) => {
		console.log('*** drag start', e)
		this.setState({dragEndTransition: 'none'})
	}

	handleDrag = (e, data) => {
		this.setState({isDragging: true});
	}

	handleDragStop = (e, data) => {
		const { isOpen } = this.state;

		const touchPos = data.y

		console.log('*** drag end', touchPos, WINDOW_TOP)
		if (!isOpen & touchPos < TRIGGER_DRAG_DISTANCE || isOpen & touchPos < TRIGGER_DRAG_DISTANCE * 2) {
			this.setState({
				isOpen: true,
				isDragging: false,
				position: WINDOW_TOP + this.context.dimensions.player.height,
				dragEndTransition: DRAG_TRANSITION,
			});
			return console.log('UP')
		}
		this.setState({
			isOpen: false,
			isDragging: false,
			position: 0,
			dragEndTransition: DRAG_TRANSITION,
		});
		console.log('DOWN')
		// return true;
	}

	handlePlayerToggle = () => {
		const { isOpen } = this.state;

		if (!isOpen) {
			this.setState({
				isOpen: true,
				position: WINDOW_TOP + this.context.dimensions.player.height,
			});
			return console.log('UP')
		}
		this.setState({
			isOpen: false,
			position: 0,
		});
		console.log('DOWN')
	}

	handleQueueToggle = () => {
		console.log('handleQueueToggle')
		this.setState({showQueue: !this.state.showQueue})
	}

	render() {
		const { 
			player, 
			userAgentIsMobile,
			handleStopTrack,
			handlePlayTrack,
			handlePauseTrack,
			handlePlayNext,
			handlePlayPrev,
			handlePlayFromQueue,
			handleSeek,
			classes,
		} = this.props;

		const {
			isOpen,
			isDragging,
			showQueue,
			position,
			windowHeight,
			dragEndTransition,
		} = this.state;

		const theme = this.context;

		console.log('isOpen:', isOpen)
		return (
			<div 
				style={{zIndex: isOpen || isDragging ? 1 : 0}}
				className={classes.root}
			>
				{ !userAgentIsMobile && 
					<Backdrop open={isOpen} onBackdropClick={this.handlePlayerToggle} />
				}
				<Draggable
					// disabled={!userAgentIsMobile}
					axis="y"
					defaultPosition={{x: 0, y: 0}}
	        position={{x: 0, y: position}}
	        scale={1}
					onStart={this.handleDragStart}
					onDrag={this.handleDrag}
					onStop={this.handleDragStop}
				>
					<div 
						style={{transition: dragEndTransition}}
						className={userAgentIsMobile ? classes.containerMobile : classes.containerDesktop}
					>
						<PlayerBar
							player={player}
							playerIsOpen={isOpen}
							showQueue={showQueue}
							userAgentIsMobile={userAgentIsMobile}
							handleStopTrack={handleStopTrack}
							handlePlayTrack={handlePlayTrack}
							handlePauseTrack={handlePauseTrack}
							handlePlayNext={handlePlayNext}
							handlePlayPrev={handlePlayPrev}
							handlePlayerToggle={this.handlePlayerToggle}
							handleQueueToggle={this.handleQueueToggle}
						/>
						<div style={{
							background: theme.palette.secondary.light,
							position: 'absolute',
							height: windowHeight - theme.dimensions.player.height,
							width: '100%',
						}}>
							{ showQueue ?
								<QueueList 
									queue={player.queue}
									playerIsOpen={isOpen}
									queueIndex={player.queueIndex}
									currentTrack={player.currentTrack} 
									handleStopTrack={handleStopTrack}
									handlePlayTrack={handlePlayTrack}
									handlePlayFromQueue={handlePlayFromQueue}
								/>
								:
								<div style={{display: 'flex'}}>
									<img 
										src={player.currentTrack.image.src} 
										alt={`Album art for ${player.currentTrack.album.title}`}
										width={theme.dimensions.player.width}
										height={userAgentIsMobile ? window.innerWidth : theme.dimensions.player.width}
									/>
								</div>
							}
						</div>
						<div style={{
							backgroundColor: getSecondaryBackgroundColor({theme}), 
							position: 'fixed',
							width: '100%',
							bottom: !isOpen ? theme.dimensions.player.height : (window.innerHeight - (theme.dimensions.player.height * 2)) * -1,
							zIndex: 1000,
						}}>
							<PlayerProgress 
								player={player}
								playerIsOpen={isOpen}
								handleSeek={handleSeek}
							/>
						</div>
						{ isOpen &&
							<div style={{
									backgroundColor: getSecondaryBackgroundColor({theme}), 
									position: 'fixed',
									width: '100%',
									bottom: (window.innerHeight - theme.dimensions.player.height) * -1,
								}}
							>
								<PlayerControls
									player={player}
									playerIsOpen={isOpen}
									userAgentIsMobile={userAgentIsMobile}
									handlePlayTrack={handlePlayTrack}
									handlePauseTrack={handlePauseTrack}
									handlePlayNext={handlePlayNext}
									handlePlayPrev={handlePlayPrev}
									handlePlayerToggle={this.handlePlayerToggle}
								/>
							</div>
						}
					</div>
				</Draggable>
			</div>
		);
	}
}

Player.propTypes = {
	player: PropTypes.object.isRequired,
	userAgentIsMobile: PropTypes.bool.isRequired,
	handlePlayTrack: PropTypes.func.isRequired,
	handlePauseTrack: PropTypes.func.isRequired,
	handleStopTrack: PropTypes.func.isRequired,
	handlePlayNext: PropTypes.func.isRequired,
	handlePlayPrev: PropTypes.func.isRequired,
	handlePlayFromQueue: PropTypes.func.isRequired,
	handleSeek: PropTypes.func.isRequired,
}

export default withStyles(styles)(Player);
