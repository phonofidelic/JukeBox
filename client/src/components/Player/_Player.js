import React, { Component, createRef } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import posed from 'react-pose';

import {
  ThemeContext,
  getSecondaryBackgroundColor,
  getPlayerHeight,
  getPlayerWidth,
  getNavMobileHeight
} from '../../contexts/theme.context';
import Backdrop from '../Backdrop';

import PlayerProgress from 'components/Player/PlayerProgress';
import PlayerControls from 'components/Player/PlayerControls';
import QueueList from 'components/Player/QueueList';
import CurrentTrack from 'components/Player/CurrentTrack';
import ToggleQueueButton from 'components/Player/ToggleQueueButton';

// import Draggable from 'react-draggable';
import ReactCardFlip from 'react-card-flip';

// const WINDOW_TOP = window.innerHeight * -1;
const TRIGGER_DRAG_DISTANCE = window.innerHeight / 4;
const DRAG_TRANSITION = 'all .5s ease';

// const poseConfig = {
// 	draggable: 'y',
//   dragEnd: {
//     transition: DRAG_TRANSITION,
//     // Player position snaps back to initial pos if isOpen change is not triggered.
//     // Otherwise, open/closed props below override this setting.
//     y: props => props.isOpen ? 0 : window.innerHeight - getPlayerHeight(props),
//   },
//   open: { y: 0 },
//   closed: { y: props => window.innerHeight - getPlayerHeight(props)},
// };

const Container = styled.div`
	position: fixed;
	width: ${props =>
    props.userAgentIsMobile ? '100%' : `${getPlayerWidth(props)}px`};
	margin: 0 auto;
	// top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	display: flex;
	z-index: 2;
	// height: ${props => (props.isOpen ? '100%' : getPlayerHeight(props) + 'px')};
	transition: ${props => props.dragEndTransition};
`;

const LayoutContainer = styled.div`
  position: fixed;
  top: ${props => (props.isOpen ? 0 : props.userAgentIsMobile ? -56 : 0)}px;
  bottom: ${props =>
    props.userAgentIsMobile ? getNavMobileHeight(props) * 2 : 0}px;
  background-color: ${getSecondaryBackgroundColor};
  box-shadow: ${props =>
    !props.userAgentIsMobile && props.isOpen
      ? 'none'
      : '0px 5px 10px 1px #000'};
  width: ${props =>
    props.userAgentIsMobile ? '100%' : `${getPlayerWidth(props)}px`};
  z-index: 1;
  display: flex;
  flex-direction: column;
`;

const DraggableLayoutContainer = posed(LayoutContainer)({
  // draggable: ({ draggable }) => draggable,
  draggable: 'y',
  dragBounds: {
    top: 0,
    bottom: props => window.innerHeight - getPlayerHeight(props)
  },
  dragEnd: {
    // Player position snaps back to initial pos if isOpen change is not triggered.
    // Otherwise, open/closed props below override this setting.
    y: props =>
      props.isOpen ? 0 : window.innerHeight - getPlayerHeight(props),
    transition: DRAG_TRANSITION
  },
  open: {
    y: 0,
    transition: DRAG_TRANSITION
  },
  closed: {
    y: props => window.innerHeight - getPlayerHeight(props),
    transition: DRAG_TRANSITION
  }
  // closed: { y: 56 },
  // label: 'layout-container',
  // pressable: ({ isMobile }) => isMobile ? true : false,
  // init: { scale: 1 },
  // press: { scale: 1.1 }
});

const TopPanel = styled.div`
  background-color: ${getSecondaryBackgroundColor};
  // background-color: green;
  // border-bottom: 1px solid #dcdcdc;
  display: flex;
  width: 100%;
  margin: 0 auto;
`;

// const DraggableTopPanel = posed(TopPanel)({
// 	draggable: 'y',
// 	dragBounds: {
// 		top: 0,
// 		bottom: props => window.innerHeight - getPlayerHeight(props)
// 	},
//   dragEnd: {
//     transition: { type: 'spring' },
//     // Player position snaps back to initial pos if isOpen change is not triggered.
//     // Otherwise, open/closed props below override this setting.
//     y: props => props.isOpen ? 0 : window.innerHeight - getPlayerHeight(props),
//   },
//   open: { y: 0 },
//   closed: { y: props => window.innerHeight - getPlayerHeight(props)},
// })

const QueueContainer = styled.div`
	width: ${props =>
    props.userAgentIsMobile ? '100%' : `${getPlayerWidth(props)}px`};
	// transition: ${DRAG_TRANSITION};
	// position: fixed;
  // top: ${props =>
    props.isOpen ? getPlayerHeight(props) + 'px' : window.innerHeight};
  height: 1px;
  flex: 1;
	z-index: 3;
`;
const PosedQueueContainer = posed(QueueContainer)({
  draggable: false
});

const PlayerProgressContainer = styled.div`
	//background-color: ${getSecondaryBackgroundColor};
	// position: fixed;
	// top: ${props => (props.isOpen || !props.userAgentIsMobile ? null : 0)};
	top: ${props =>
    props.userAgentIsMobile
      ? props.isOpen
        ? null
        : 0
      : props.isOpen
      ? null
      : 0};
	bottom: ${props =>
    props.userAgentIsMobile
      ? props.isOpen
        ? `${getNavMobileHeight(props) * 2}px`
        : null
      : props.isOpen
      ? `${getNavMobileHeight(props)}px`
      : null};
	width: ${props =>
    props.userAgentIsMobile ? '100%' : `${getPlayerWidth(props)}px`};
	z-index: 1000;
	transition: ${props => props.dragEndTransition};
`;

const BottomPanel = styled.div`
  background-color: ${props =>
    props.isOpen ? getSecondaryBackgroundColor(props) : 'none'};
  display: flex;
  position: fixed;
  top: ${props => (props.isOpen ? null : 0)};
  bottom: ${props =>
    props.userAgentIsMobile ? getNavMobileHeight(props) : 0}px;
  width: ${props =>
    props.userAgentIsMobile ? '100%' : `${getPlayerWidth(props)}px`};
  // margin-left: auto;
  z-index: 3;
`;

export class Player extends Component {
  static contextType = ThemeContext;

  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
      showQueue: true,
      // position: 0,
      windowHeight: window.innerHeight,
      dragEndTransition: 'none'
      // isDraggable: false
      // isDraggable: true
    };
  }

  componentDidMount() {
    window.addEventListener('resize', this.handleWindowResize.bind(this));
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleWindowResize);
  }

  handleWindowResize(e) {
    const theme = this.context;
    console.log(
      'QueueContainer height:',
      e.target.innerHeight - getPlayerHeight({ theme })
    );
    this.setState({ windowHeight: e.target.innerHeight });
  }

  handleDragStart = e => {
    // e.preventDefault()
    if (!e.changedTouches) return;

    const dragStartPos = e.changedTouches[0].pageY;
    const theme = this.context;
    const { isOpen } = this.state;

    console.log('Player, handleDragstart, e:', e);
    console.log('dragStartPos:', dragStartPos);

    // If drag start position is below the top panel,
    // ignore and do not drag
    if (isOpen & (dragStartPos > getPlayerHeight({ theme }))) {
      console.log("*** DON'T DRAG ***");
      e.preventDefault();
      return this.setState({ isDraggable: false });
    }

    console.log('handleDragStart, e', dragStartPos);
    this.setState({
      ...this.state,
      dragStartPos: dragStartPos,
      isDraggable: 'y'
    });
  };

  handleDrag = (e, data) => {
    // console.log('Player, handleDrag, e:', e.changedTouches[0].clientY)
  };

  handleDragStop = e => {
    if (!e.changedTouches) return;
    console.log('handleDragStop, e', e.changedTouches[0].pageY);
    const { isOpen } = this.state;
    const theme = this.context;
    const { dragStartPos } = this.state;
    const dragEndPos = e.changedTouches[0].pageY;

    let touchDelta = dragStartPos - dragEndPos;
    if (touchDelta < 0) touchDelta = touchDelta * -1;
    console.log('touchDelta:', touchDelta);
    if (touchDelta < TRIGGER_DRAG_DISTANCE) return;

    // If player is open and dragStartPos is within the queue list area
    if (isOpen & (dragStartPos > getPlayerHeight({ theme }))) return;

    console.log('*** drag end', dragEndPos, TRIGGER_DRAG_DISTANCE);
    if (
      !isOpen & (dragEndPos < window.innerHeight - TRIGGER_DRAG_DISTANCE) ||
      isOpen & (dragEndPos < TRIGGER_DRAG_DISTANCE)
    ) {
      this.setState({
        isOpen: true
        // translatePos: 0,
      });
      return console.log('UP');
    }
    this.setState({
      isOpen: false
    });
    console.log('DOWN');
  };

  handlePlayerToggle = () => {
    const { isOpen } = this.state;
    const theme = this.context;

    if (!isOpen) {
      this.setState({
        isOpen: true,
        position: getPlayerHeight({ theme }) * -1,
        dragEndTransition: DRAG_TRANSITION
      });
      return console.log('UP');
    }
    this.setState({
      isOpen: false,
      position: 0,
      // position: getPlayerHeight({theme}),
      dragEndTransition: DRAG_TRANSITION
    });
    console.log('DOWN');
  };

  handleQueueToggle = () => {
    console.log('handleQueueToggle');
    this.setState({ showQueue: !this.state.showQueue });
  };

  render() {
    const {
      player,
      userAgentIsMobile,
      handleStopTrack,
      handlePlayTrack,
      handlePauseTrack,
      handlePlayNext,
      handlePlayPrev,
      handlePlayFromQueue,
      handleSeek
      // classes,
    } = this.props;

    const {
      isOpen,
      showQueue,
      // position,
      windowHeight,
      dragEndTransition,
      isDraggable
    } = this.state;

    const theme = this.context;

    return (
      <Container
        theme={theme}
        isOpen={isOpen}
        windowHeight={windowHeight}
        userAgentIsMobile={userAgentIsMobile}
      >
        {!userAgentIsMobile && (
          <Backdrop open={isOpen} onBackdropClick={this.handlePlayerToggle} />
        )}
        <DraggableLayoutContainer
          id="player_layout-container"
          theme={theme}
          userAgentIsMobile={userAgentIsMobile}
          isOpen={isOpen}
          onDragStart={this.handleDragStart}
          onDragEnd={this.handleDragStop}
          //dragEndTransition={'y'}
          dragBounds={{
            top: 0,
            bottom: window.innerHeight - getPlayerHeight({ theme })
          }}
          pose={isOpen ? 'open' : 'closed'}
          draggable={isDraggable}
        >
          <TopPanel id="player_top-panel" theme={theme} isOpen={isOpen}>
            <CurrentTrack player={player} playerIsOpen={isOpen} />
            {isOpen && (
              <ToggleQueueButton
                showQueue={showQueue}
                handleQueueToggle={this.handleQueueToggle}
              />
            )}
          </TopPanel>

          <QueueContainer
            id="player_queue-container"
            theme={theme}
            isOpen={isOpen}
            userAgentIsMobile={userAgentIsMobile}
            windowHeight={windowHeight}
          >
            <ReactCardFlip
              //type="horizontal"
              isFlipped={!showQueue}
              //style={{width: '100%'}}
            >
              <QueueList
                key="front"
                queue={player.queue}
                isOpen={isOpen}
                playerIsOpen={isOpen}
                windowHeight={windowHeight}
                queueIndex={player.queueIndex}
                currentTrack={player.currentTrack}
                handleStopTrack={handleStopTrack}
                handlePlayTrack={handlePlayTrack}
                handlePlayFromQueue={handlePlayFromQueue}
              />
              <div key="back" style={{ display: 'flex' }}>
                <img
                  src={player.currentTrack.image.src}
                  alt={`Album art for ${player.currentTrack.album.title}`}
                  width={
                    userAgentIsMobile
                      ? window.innerWidth
                      : getPlayerWidth({ theme })
                  }
                  height={
                    userAgentIsMobile
                      ? window.innerWidth
                      : getPlayerWidth({ theme })
                  }
                />
              </div>
            </ReactCardFlip>
          </QueueContainer>

          <PlayerProgressContainer
            id="player_progress-container"
            theme={theme}
            userAgentIsMobile={userAgentIsMobile}
            isOpen={isOpen}
            windowHeight={windowHeight}
            dragEndTransition={dragEndTransition}
          >
            <PlayerProgress
              player={player}
              playerIsOpen={isOpen}
              userAgentIsMobile={userAgentIsMobile}
              handleSeek={handleSeek}
            />
          </PlayerProgressContainer>
          <BottomPanel
            id="player_bottom-panel"
            theme={theme}
            isOpen={isOpen}
            windowHeight={windowHeight}
            userAgentIsMobile={userAgentIsMobile}
          >
            {userAgentIsMobile && !isOpen && (
              <div style={{ display: 'flex', width: '100%' }}></div>
            )}
            <PlayerControls
              player={player}
              playerIsOpen={isOpen}
              userAgentIsMobile={userAgentIsMobile}
              handlePlayTrack={handlePlayTrack}
              handlePauseTrack={handlePauseTrack}
              handlePlayNext={handlePlayNext}
              handlePlayPrev={handlePlayPrev}
              handlePlayerToggle={this.handlePlayerToggle}
            />
          </BottomPanel>
        </DraggableLayoutContainer>

        {/* <QueueContainer
          id="player_queue-container"
          theme={theme}
          isOpen={isOpen}
          userAgentIsMobile={userAgentIsMobile}
          windowHeight={windowHeight}
        >
          <ReactCardFlip
            //type="horizontal"
            isFlipped={!showQueue}
            //style={{width: '100%'}}
          >
            <QueueList
              key="front"
              queue={player.queue}
              isOpen={isOpen}
              playerIsOpen={isOpen}
              windowHeight={windowHeight}
              queueIndex={player.queueIndex}
              currentTrack={player.currentTrack}
              handleStopTrack={handleStopTrack}
              handlePlayTrack={handlePlayTrack}
              handlePlayFromQueue={handlePlayFromQueue}
            />
            <div key="back" style={{ display: 'flex' }}>
              <img
                src={player.currentTrack.image.src}
                alt={`Album art for ${player.currentTrack.album.title}`}
                width={
                  userAgentIsMobile
                    ? window.innerWidth
                    : getPlayerWidth({ theme })
                }
                height={
                  userAgentIsMobile
                    ? window.innerWidth
                    : getPlayerWidth({ theme })
                }
              />
            </div>
          </ReactCardFlip>
        </QueueContainer> */}
      </Container>
    );
  }
}

Player.propTypes = {
  player: PropTypes.object.isRequired,
  userAgentIsMobile: PropTypes.bool.isRequired,
  handlePlayTrack: PropTypes.func.isRequired,
  handlePauseTrack: PropTypes.func.isRequired,
  handleStopTrack: PropTypes.func.isRequired,
  handlePlayNext: PropTypes.func.isRequired,
  handlePlayPrev: PropTypes.func.isRequired,
  handlePlayFromQueue: PropTypes.func.isRequired,
  handleSeek: PropTypes.func.isRequired
};

export default Player;
