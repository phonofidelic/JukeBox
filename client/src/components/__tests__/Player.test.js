import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { createRender } from '@material-ui/core/test-utils';
import Player from '../Player';
import { INITIAL_STATE } from '../../reducers/player.reducer';

describe('Player', () => {
	let props;

	beforeEach(() => {
		props = {
			player: INITIAL_STATE,
			selectedTrack: null,
			userAgentIsMobile: false,
			handlePlayTrack: jest.fn(),
			handlePauseTrack: jest.fn(),
			handleStopTrack: jest.fn(),
			handlePlayNext: jest.fn(),
			handlePlayPrev: jest.fn(),
			handleToggleQueue: jest.fn(),
			handlePlayFromQueue: jest.fn(),
			handleSeek: jest.fn(),
		}
	});

	it('renders correctly', () => {
		const tree = renderer
			.create( <Player {...props} />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});