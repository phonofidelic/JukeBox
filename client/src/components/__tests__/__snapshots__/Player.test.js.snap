// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#ff4081",
      "borderBottom": "solid 1px #3f51b5",
      "bottom": 0,
      "boxShadow": "0px -1px 20px 1px #ccc",
      "padding": "0",
      "position": "fixed",
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <div
    style={
      Object {
        "background": "#ccc",
        "height": "10px",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "background": "#f50057",
          "height": "100%",
          "width": 0,
        }
      }
    />
    <div
      style={
        Object {
          "display": "flex",
          "height": "10px",
          "position": "absolute",
          "top": "0",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-39"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-39"
        >
          <button
            className="MuiButtonBase-root-98 MuiIconButton-root-92"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-97"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-101"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M6 6h2v12H6zm3.5 6l8.5 6V6z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-107"
            />
          </button>
          <button
            className="MuiButtonBase-root-98 MuiIconButton-root-92"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-97"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-101"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M8 5v14l11-7z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-107"
            />
          </button>
          <button
            className="MuiButtonBase-root-98 MuiIconButton-root-92"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-97"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-101"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M6 18l8.5-6L6 6v12zM16 6v12h2V6h-2z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-107"
            />
          </button>
        </div>
        <div
          style={
            Object {
              "position": "fixed",
              "right": "0px",
              "zIndex": "1001",
            }
          }
        >
          <button
            className="MuiButtonBase-root-98 MuiIconButton-root-92"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-97"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-101"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-107"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiCollapse-container-114 MuiCollapse-entered-115"
    direction="up"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper-116"
    >
      <div
        className="MuiCollapse-wrapperInner-117"
      >
        <ul
          className="MuiList-root-118 MuiList-padding-119"
          style={
            Object {
              "padding": "0",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
